### ISSUE TITLE
WARNING: received a nan loss, setting loss value to 0

### ISSUE THREAD

**Issue by bigcash** [SeanNaren/deepspeech.pytorch#485]
i try to train by dataset "st_cmds", and before 20 epoch is working, and get nan loss after "WARNING: received a nan loss, setting loss value to 0", how i keep on training it, skip this problem?
thanks

100%|██████████| 209/209 [02:03<00:00,  1.69it/s]Validation Summary Epoch: [20] Average WER 20.853      Average CER 20.861
...
Epoch: [21][1390/1930]  Time 0.690 (0.986)      Data 0.004 (0.009)      Loss 2.5681 (10.9521)
Epoch: [21][1391/1930]  Time 0.782 (0.986)      Data 0.004 (0.009)      Loss 4.3603 (10.9519)
Epoch: [21][1392/1930]  Time 0.758 (0.986)      Data 0.004 (0.009)      Loss 3.0738 (10.9517)
Epoch: [21][1393/1930]  Time 0.699 (0.986)      Data 0.004 (0.009)      Loss 3.9250 (10.9515)
Epoch: [21][1394/1930]  Time 0.776 (0.986)      Data 0.004 (0.009)      Loss 2.5246 (10.9513)
Epoch: [21][1395/1930]  Time 0.882 (0.986)      Data 0.004 (0.009)      Loss 4.2818 (10.9511)
Epoch: [21][1396/1930]  Time 0.722 (0.986)      Data 0.005 (0.009)      Loss 3.6416 (10.9510)
Epoch: [21][1397/1930]  Time 0.780 (0.986)      Data 0.003 (0.009)      Loss 5.0185 (10.9508)
Epoch: [21][1398/1930]  Time 0.765 (0.986)      Data 0.004 (0.009)      Loss 2.9142 (10.9506)
Epoch: [21][1399/1930]  Time 0.863 (0.986)      Data 0.005 (0.009)      Loss 4.4609 (10.9505)
Epoch: [21][1400/1930]  Time 0.754 (0.986)      Data 0.005 (0.009)      Loss 4.2682 (10.9503)
Epoch: [21][1401/1930]  Time 0.811 (0.986)      Data 0.004 (0.009)      Loss 3.6588 (10.9501)
Epoch: [21][1402/1930]  Time 0.785 (0.986)      Data 0.004 (0.009)      Loss 3.0358 (10.9499)
Epoch: [21][1403/1930]  Time 0.769 (0.986)      Data 0.003 (0.009)      Loss 3.0952 (10.9497)
Epoch: [21][1404/1930]  Time 0.747 (0.986)      Data 0.004 (0.009)      Loss 3.7729 (10.9495)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [21][1405/1930]  Time 0.325 (0.986)      Data 0.004 (0.009)      Loss 0.0000 (10.9493)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [21][1406/1930]  Time 0.314 (0.986)      Data 0.006 (0.009)      Loss 0.0000 (10.9490)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [21][1407/1930]  Time 0.288 (0.986)      Data 0.005 (0.009)      Loss 0.0000 (10.9487)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [21][1408/1930]  Time 0.318 (0.986)      Data 0.005 (0.009)      Loss 0.0000 (10.9484)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [21][1409/1930]  Time 0.315 (0.986)      Data 0.005 (0.009)      Loss 0.0000 (10.9482)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [21][1410/1930]  Time 0.298 (0.986)      Data 0.006 (0.009)      Loss 0.0000 (10.9479)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [21][1411/1930]  Time 0.306 (0.986)      Data 0.006 (0.009)      Loss 0.0000 (10.9476)

WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [21][1928/1930]  Time 0.274 (0.977)      Data 0.005 (0.009)      Loss 0.0000 (10.8079)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [21][1929/1930]  Time 0.304 (0.977)      Data 0.005 (0.009)      Loss 0.0000 (10.8076)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [21][1930/1930]  Time 0.335 (0.977)      Data 0.005 (0.009)      Loss 0.0000 (10.8074)
Training Summary Epoch: [21]    Time taken (s): 1272    Average Loss 1.535
  9%|▊         | 18/209 [00:08<01:29,  2.11it/s]
 17%|█▋        | 36/209 [00:16<01:16,  2.26it/s]
 25%|██▌       | 53/209 [00:24<01:07,  2.30it/s]
 33%|███▎      | 70/209 [00:32<01:02,  2.22it/s]
 49%|████▉     | 102/209 [00:46<00:45,  2.34it/s]
 70%|███████   | 147/209 [01:06<00:29,  2.11it/s]
 91%|█████████ | 189/209 [01:24<00:08,  2.27it/s]
 97%|█████████▋| 203/209 [01:30<00:03,  2.30it/s]
100%|██████████| 209/209 [01:33<00:00,  2.23it/s]Validation Summary Epoch: [21] Average WER 100.000     Average CER 100.000
Learning rate annealed to: 0.000243
Shuffling batches...

WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [22][1/1930]     Time 1.826 (0.977)      Data 1.465 (0.009)      Loss 0.0000 (10.8071)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [22][2/1930]     Time 0.331 (0.977)      Data 0.029 (0.009)      Loss 0.0000 (10.8068)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [22][3/1930]     Time 0.068 (0.977)      Data 0.004 (0.009)      Loss 0.0000 (10.8068)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [22][4/1930]     Time 0.266 (0.977)      Data 0.001 (0.009)      Loss 0.0000 (10.8065)

**Comment by derek-sq** [comment]
Same issue. My data input have Nan, so you can print your input every batch..

**Comment by bigcash** [comment]
it happened after epoch 10,  when loss is very small value, not because of data

**Comment by kouohhashi** [comment]
Same here. Have you found the solution?

**Comment by tingyang01** [comment]
Same here, I've got this error.
received a nan loss, setting loss value to 0.
Do you have any solution for it?

**Comment by tingyang01** [comment]
Epoch: [1][4686/6051]   Time 1.781 (1.857)      Data 0.005 (0.005)      Loss 42.8735 (73.3981)
Epoch: [1][4687/6051]   Time 1.501 (1.857)      Data 0.005 (0.005)      Loss 44.6281 (73.3920)
Epoch: [1][4688/6051]   Time 1.752 (1.857)      Data 0.004 (0.005)      Loss 53.4758 (73.3877)
Epoch: [1][4689/6051]   Time 1.869 (1.857)      Data 0.005 (0.005)      Loss 44.8665 (73.3817)
Epoch: [1][4690/6051]   Time 1.800 (1.857)      Data 0.005 (0.005)      Loss 45.0176 (73.3756)
Epoch: [1][4691/6051]   Time 1.861 (1.857)      Data 0.005 (0.005)      Loss 40.6438 (73.3686)
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
Epoch: [1][4692/6051]   Time 0.546 (1.856)      Data 0.004 (0.005)      Loss 0.0000 (73.3530)
Epoch: [1][4693/6051]   Time 1.667 (1.856)      Data 0.005 (0.005)      Loss 40.4348 (73.3460)
Epoch: [1][4694/6051]   Time 1.924 (1.856)      Data 0.005 (0.005)      Loss 53.1464 (73.3417)
Epoch: [1][4695/6051]   Time 1.787 (1.856)      Data 0.005 (0.005)      Loss 46.1473 (73.3359)

**Comment by liudq521** [comment]
Epoch: [25][25916/12068]	Time 0.672 (0.812)	Data 0.014 (0.101)	Loss 0.0000 (1.8408)	
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
************* 0.0
Epoch: [25][25917/12068]	Time 0.596 (0.812)	Data 0.005 (0.101)	Loss 0.0000 (1.8395)	
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
************* 0.0
Epoch: [25][25918/12068]	Time 0.610 (0.812)	Data 0.005 (0.101)	Loss 0.0000 (1.8382)	
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
************* 0.0
Epoch: [25][25919/12068]	Time 0.680 (0.812)	Data 0.012 (0.101)	Loss 0.0000 (1.8369)	
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
************* 0.0
Epoch: [25][25920/12068]	Time 0.597 (0.812)	Data 0.005 (0.101)	Loss 0.0000 (1.8356)	
torch.cuda.empty_cache()
WARNING: received a nan loss, setting loss value to 0
Skipping grad update
************* 0.0
Epoch: [25][25921/12068]	Time 1.424 (0.812)	Data 0.747 (0.102)	Loss 0.0000 (1.8343)



why?
